reconcile/terraform_resources.py

```diff
@@ -533,49 +533,52 @@ def runner(
 
 def early_exit_desired_state(*args: Any, **kwargs: Any) -> dict[str, Any]:
     gqlapi = gql.get_api()
-    namespaces = get_tf_namespaces()
-    resources = []
-    for ns_info in namespaces:
+    state_for_accounts = {
+        account["name"]: {
+            "meta": account,
+            "specs": {},
+        }
+        for account in queries.get_aws_accounts(terraform_state=True)
+    }
+    for ns_info in get_tf_namespaces():
         for spec in get_external_resource_specs(
             ns_info.dict(by_alias=True), provision_provider=PROVIDER_AWS
         ):
+            spec_resources: dict[str, str] = {}
+            spec_state = {
+                "spec": asdict(spec),
+                "resources": spec_resources,
+            }
```

Comment:
Since we are refactoring here, how about do it using comprehensions to avoid mutation? like this:

```python
            resource_paths = [
                spec.resource.get("defaults"),
                spec.resource.get("parameter_group"),
            ] + [
                spec_item.get("defaults")
                for spec_item in spec.resource.get("specs") or []
            ]
            resources = {
                resource["path"]: resource["sha256sum"]
                for path in resource_paths
                if path and (resource := gqlapi.get_resource(path))
            }
            spec_state = {
                "spec": asdict(spec),
                "resources": resources,
            }
            spec_id = f"{spec.cluster_name}/{spec.namespace_name}/{spec.provisioner_name}/{spec.provider}/{spec.identifier}"
            state_for_accounts[spec.provisioner_name][spec_id] = spec_state
```

Reply:
fixed